File: Checkerboard/src/cc/lib/checkerboard/Player.java
package cc.lib.checkerboard;

import java.util.ArrayList;
import java.util.List;

import cc.lib.game.Utils;
import cc.lib.utils.Reflector;

public class Player extends Reflector<Player> {
    static {
        addAllFields(Player.class);
    }

    int playerNum;
    Color color;

    void newGame() {

    }

    /**
     * Override to customize behavior. Default behavior is random.
     *
     * @param game
     * @param pieces
     * @return
     */
    public Piece choosePieceToMove(Game game, List<Piece> pieces) {
        return pieces.get(Utils.rand() % pieces.size());
    }

    /**
     * Override to customize behavior. Default behavior is random.
     *
     * @param game
     * @param moves
     * @return
     */
    public Move chooseMoveForPiece(Game game, List<Move> moves) {
        return moves.get(Utils.rand() % moves.size());
    }

    /**
     *
     * @return
     */
    public final Color getColor() {
        return color;
    }

    /**
     *
     * @return
     */
    public final int getPlayerNum() {
        return playerNum;
    }

}

File: Checkerboard/src/cc/lib/checkerboard/Player.kt
package cc.lib.checkerboard

import cc.lib.game.Utils
import cc.lib.reflector.Reflector

open class Player(var playerNum: Int = -1, var color: Color = Color.BLACK) : Reflector<Player>() {
	companion object {
		init {
			addAllFields(Player::class.java)
		}
	}

	fun newGame() {}

	/**
	 * Override to customize behavior. Default behavior is random.
	 *
	 * @param game
	 * @param pieces
	 * @return
	 */
	open fun choosePieceToMove(game: Game, pieces: List<Piece>): Piece? {
		return pieces[Utils.rand() % pieces.size]
	}

	/**
	 * Override to customize behavior. Default behavior is random.
	 *
	 * @param game
	 * @param moves
	 * @return
	 */
	open fun chooseMoveForPiece(game: Game, moves: List<Move>): Move? {
		return moves[Utils.rand() % moves.size]
	}
}